src/Sylius/Component/Payment/Repository/PaymentRequestRepositoryInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Payment\Repository;
+
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+
+/**
+ * @template T of PaymentRequestInterface
+ *
+ * @extends RepositoryInterface<T>
+ */
+interface PaymentRequestRepositoryInterface extends RepositoryInterface
+{
+    /**
+     * @return PaymentRequestInterface[]
+     */
+    public function findOtherExisting(PaymentRequestInterface $paymentRequest): array;
```

Comment:
```suggestion
    public function duplicateExists(PaymentRequestInterface $paymentRequest): bool;
```
Since this is (so far) only used in checking and it'd be more performant.

Reply:
I agree, it was here to be able to do more with it in the futur without changing the repository signature.